packages/@uppy/transloadit/src/index.ts

```diff
@@ -441,20 +441,14 @@ export default class Transloadit<
         // TODO: this should not live inside a `file-removed` event but somewhere more deterministic.
         // Such as inside the function where the assembly has succeeded or cancelled.
         // For the use case of cancelling the assembly when needed, we should try to do that with just `cancel-all`.
-        const fileRemovedHandler = (
-          fileRemoved: UppyFile<M, B>,
-          reason?: string,
-        ) => {
+        const fileRemovedHandler = (fileRemoved: UppyFile<M, B>) => {
           // If the assembly has successfully completed, we do not need these checks.
           // Otherwise we may cancel an assembly after it already succeeded
           if (assembly.status?.ok === 'ASSEMBLY_COMPLETED') {
             this.uppy.off('file-removed', fileRemovedHandler)
             return
           }
-          if (reason === 'cancel-all') {
-            assembly.close()
```

Comment:
Since this is on the `4.x` branch, we can also leave it removed for now and fix this in #5158, where assembly handling is already changing and it would have to be done different there after the refactor.